### PR TITLE
Fix documentation typos

### PR DESCRIPTION
**Pull request by isavynskyi**
### Short description 📝

While reading the documentation, I notice some typos and broken links. This PR resolves them. 

### Contributor checklist ✅

- [ ] The code has been linted using run `mise run lint:fix`
- [ ] The change is tested via unit testing or acceptance testing, or both
- [x] The title of the PR is formulated in a way that is usable as a changelog entry
- [ ] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [ ] The code architecture and patterns are consistent with the rest of the codebase
- [ ] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

### CODE REVIEW

File: docs/docs/guide/introduction/adopting-tuist/migrate-from-xcodeproj.md [hunk 1]
```diff
@@ -71,4 +71,4 @@ They allow defining your own abstractions for defining your projects, and most i
 Here's a list of community-developed tools that can aid you on the migration process:
 
 - [xcdiff](https://github.com/bloomberg/xcdiff): A tool which helps you diff xcodeproj files.
-- [Xcode Build Settings](https://xcodebuildsettings.com/): A reference for all the build settings available and their meaning.
```
Comment (isavynskyi) [inline docs/docs/guide/introduction/adopting-tuist/migrate-from-xcodeproj.md:74]: https://xcodebuildsettings.com/ ceased to exist. meanwhile Apple's provided https://developer.apple.com/documentation/xcode/build-settings-reference seems a valid alternative

File: docs/docs/guide/introduction/cost-of-convenience.md [hunk 2]
```diff
@@ -5,7 +5,7 @@ description: Learn about the cost of convenience in Xcode and how Tuist helps yo
 
 # The cost of convenience
 
-Before continuing, let's take a moment to consider the cost of convenience in [Xcode](https://www.google.com/url?sa=t&source=web&rct=j&opi=89978449&url=https://developer.apple.com/xcodenp/&ved=2ahUKEwik55Xo35-FAxVqxzgGHZAZCz4QFnoECAYQAQ&usg=AOvVaw28WTg1BlCGAf937pcwXMip).
```
Comment (isavynskyi) [inline docs/docs/guide/introduction/cost-of-convenience.md:8]: - not sure why we need the whole redirect via google.com here 🤷 
- destination https://developer.apple.com/xcodenp leads to Page Not Found 🔍

**Review by danieleformichelli — APPROVED** [review]

**Review by fortmarek — APPROVED** [review]
Thanks a lot for the typo fixes, that's really appreciated!

**Comment by fortmarek** [comment]
@all-contributors add @isavynskyi for docs

**Pull request merged**